include/merlin/core_kernels/upsert_and_evict.cuh

```diff
@@ -70,6 +70,8 @@ __global__ void upsert_and_evict_kernel_with_io_core(
     } while (occupy_result == OccupyResult::CONTINUE);
 
     if (occupy_result == OccupyResult::REFUSED) {
+      evicted_keys[key_idx] = insert_key;
+      evicted_scores[key_idx] = insert_score;
```

Comment:
So If a KV-pair with a small score than all KV-pairs in the bucket, the new KV-pair will be also ignored, right?

Reply:
Yeah, will be directly put into the evicted buffer. Let `data` itself make a decision.